stock_picking_invoice_product_group/wizards/stock_invoice_onshipping.py

```diff
@@ -0,0 +1,58 @@
+# -*- coding: utf-8 -*-
+# Copyright 2016 Alex Comba - Agile Business Group
+# License AGPL-3 - See http://www.gnu.org/licenses/agpl-3.0.html
+
+from openerp import models, api, fields
+
+
+class StockInvoiceOnshipping(models.TransientModel):
+    _inherit = 'stock.invoice.onshipping'
+
+    group_type = fields.Selection(
+        [('group_by_product', 'Group by product'),
+         ('group_by_product_category', 'Group by product category')
+         ],
+        string='Group Type',
+    )
+
+    @api.model
+    def _group_moves_by_group_type(self, moves, group_type):
+        """Return moves grouped by group_type"""
+        grouped_moves = {}
+        for move in moves:
+            partner = move.picking_id.partner_id
+            product = move.product_id
+            categ = product.categ_id
+            if group_type == 'group_by_product':
+                key = (partner.id, product.id)
+            else:
```

Comment:
Of course, I'm not suggesting to prevent the user from choosing an option instead of the other. I'm just suggesting to handle the two different cases ('group_by_product' and 'group_by_product_category') in an if/elif statement rather than in an if/else statement. Something like:
```

if group_type == 'group_by_product':
    key = (partner.id, product.id)
elif group_type == 'group_by_product_category':
    key = (partner.id, categ.id)
else:
    #Some kind of exception raising/handling
```

That's because an else statement in this case would implicitly define a default: let's assume the user doesn't select any group type in the wizard (which is currently possible, being the field not required), the system would behave exactly like if he selected the option 'group_by_product_category', regardless if he really did it or not.

If this is the desired behavior, it would be better to explicitly define that option as a default for the field. Moreover, I believe that the system should force the user to choose an option.